client-js/app/scripts/app.ts

```diff
@@ -30,6 +31,51 @@ interface Creds {
 }
 
 
+interface Services {
+  places?: Places;
+  socket: rpc.JsonRpcSocket;
+  mgr: manager.ManagerApi;
+  reg: registry.RegistryApi;
+  dv: datavault.DataVaultApi;
+  node: nodeApi.NodeApi;
+}
+
+
+/**
+ * Make a route with n path segments labeled p<i>.
+ *
+ * On a page change, the path can be extracted using the getPath function.
+ * TODO: figure out a way to use wildcard routes instead of multiple routes
+ * each with a fixed number of segments.
+ */
+function pathRoute(n: number): string {
+  var route = '';
+  for (var i = 0; i < n; i++) {
+    route += ':p' + i + '/';
```

Comment:
LOL; I changed this to use string interpolation, which is a bit less whimsical but more readable.